[GTK][WPE] Address bots for TestWebsiteData/cache flakiness

abstractmachines opened the pull request:
#### 75a3eb3a5fe3dce1867d880faf785137963b234a
<pre>
[GTK][WPE] Address bots for TestWebsiteData/cache flakiness
<a href="https://bugs.webkit.org/show_bug.cgi?id=259991">https://bugs.webkit.org/show_bug.cgi?id=259991</a>

Reviewed by Michael Catanzaro.

This is a test that succeeds locally but fails intermittently in bots per
history below, and in `wk-bot-digest` over time.

In 2018 in `webkit.org/b/188113`, we determined that this test was flaky
in GTK, and in 2019 in that bug we also thought possibly that the 1-second wait
was not long enough in this test, and perhaps we should increase the wait
to 2s. In 2022, we determined in that bug that test had been consistently
passing in GTK/WPE for a long time &quot;Sometimes fails consistently, but due
to regressions.&quot; We then removed the `&quot;all&quot;` pass/fail flakes in that commit
 `246255@main` in January 2022. Since that time the bots tend to look like
 this, despite succeeding locally for GTK|WPE:

```
/path/to/wk-bot-digest-results/last-3000 $ ag /webkit/WebKitWebsiteData/cache | sort | grep PASS | wc -l
2066
/path/to/wk-bot-digest-results/last-3000 $ ag /webkit/WebKitWebsiteData/cache | sort | grep FAIL | wc -l
75
```

In March 2022 a few months ago we re-added the failing/flaky expectation
for GTK in `webkit.org/b/254002`, so this test&apos;s failing/flaky expectations
have been continually added and removed over time.

Per comment in b/188113, adding a slightly higher wait for bots.
Removing failing expectations.

* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebsiteData.cpp:
(testWebsiteDataCache):
* Tools/TestWebKitAPI/glib/TestExpectations.json:

Canonical link: <a href="https://commits.webkit.org/266790@main">https://commits.webkit.org/266790@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c4fe4f42b1a6d77ae9fdb727bc028929e3866d82

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/14629 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/14939 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/15287 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/16377 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/13815 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/14764 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/17456 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/15022 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/16462 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/14810 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/15320 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/12427 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/17111 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/12603 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/13194 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/20205 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/13682 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/13360 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/16604 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/13913 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/11749 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/13207 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3563 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/17545 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/13756 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro reviewed (approved):
I really dislike hardcoded waits in tests, since they slow things down considerably, but it's already there and certainly better than a flaky test.

abstractmachines committed:
[GTK][WPE] Address bots for TestWebsiteData/cache flakiness
https://bugs.webkit.org/show_bug.cgi?id=259991

Reviewed by Michael Catanzaro.

This is a test that succeeds locally but fails intermittently in bots per
history below, and in `wk-bot-digest` over time.

In 2018 in `webkit.org/b/188113`, we determined that this test was flaky
in GTK, and in 2019 in that bug we also thought possibly that the 1-second wait
was not long enough in this test, and perhaps we should increase the wait
to 2s. In 2022, we determined in that bug that test had been consistently
passing in GTK/WPE for a long time "Sometimes fails consistently, but due
to regressions." We then removed the `"all"` pass/fail flakes in that commit
 `246255@main` in January 2022. Since that time the bots tend to look like
 this, despite succeeding locally for GTK|WPE:

```
/path/to/wk-bot-digest-results/last-3000 $ ag /webkit/WebKitWebsiteData/cache | sort | grep PASS | wc -l
2066
/path/to/wk-bot-digest-results/last-3000 $ ag /webkit/WebKitWebsiteData/cache | sort | grep FAIL | wc -l
75
```

In March 2022 a few months ago we re-added the failing/flaky expectation
for GTK in `webkit.org/b/254002`, so this test's failing/flaky expectations
have been continually added and removed over time.

Per comment in b/188113, adding a slightly higher wait for bots.
Removing failing expectations.

* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebsiteData.cpp:
(testWebsiteDataCache):
* Tools/TestWebKitAPI/glib/TestExpectations.json:

Canonical link: https://commits.webkit.org/266790@main
